src/ol/format/wkt.js

```diff
@@ -577,26 +589,40 @@ ol.format.WKT.Parser.prototype.parse = function() {
   return geometry;
 };
 
-
 /**
- * @return {!(ol.geom.Geometry|ol.geom.GeometryCollection)} The geometry.
+ * @return {!ol.geom.Geometry} The geometry.
  * @private
  */
 ol.format.WKT.Parser.prototype.parseGeometry_ = function() {
   var token = this.token_;
   if (this.match(ol.format.WKT.TokenType.TEXT)) {
     var geomType = token.value;
+    var dimToken = this.token_;
+    var isEmpty = false;
+    if (this.match(ol.format.WKT.TokenType.TEXT)) {
+      var dimInfo = dimToken.value;
+      if (dimInfo === 'Z') {
+        this.layout_ = ol.geom.GeometryLayout.XYZ;
+      } else if (dimInfo === 'M') {
+        this.layout_ = ol.geom.GeometryLayout.XYM;
+      } else if (dimInfo === 'ZM') {
+        this.layout_ = ol.geom.GeometryLayout.XYZM;
+      } else if (dimInfo === 'EMPTY') {
```

Comment:
Ah I see your comment now (It's handled in the parsers but if dimInfo is found the text is already consumed.) which explains why this isn't possible